examples/Services/Planning/AddKeywordPlan.cs

```diff
@@ -0,0 +1,322 @@
+        }
+
+        /// <summary>
+        /// Creates the campaign for the keyword plan.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        /// <param name="keywordPlanResource">The keyword plan resource.</param>
+        /// <returns>The newly created campaign resource.</returns>
+        private string CreateKeywordPlanCampaign(GoogleAdsClient client, long customerId,
+            String keywordPlanResource)
+        {
+            // Get the KeywordPlanCampaignService.
+            KeywordPlanCampaignServiceClient serviceClient = client.GetService(
+                Services.V0.KeywordPlanCampaignService);
+
+            // Create a keyword plan campaign.
+            KeywordPlanCampaign campaign = new KeywordPlanCampaign()
+            {
+                Name = "Keyword plan campaign #" + ExampleUtilities.GetRandomString(),
+                CpcBidMicros = 1_000_000L,
+                KeywordPlanNetwork = KeywordPlanNetwork.GoogleSearch,
+                KeywordPlan = keywordPlanResource
+            };
+
+            // See https://developers.google.com/adwords/api/docs/appendix/geotargeting
+            // for the list of geo target IDs.
+            campaign.GeoTargets.Add(new KeywordPlanGeoTarget()
+            {
+                GeoTargetConstant = ResourceNames.GeoTargetConstant(2840) /* USA */
+            });
+
+            // See https://developers.google.com/adwords/api/docs/appendix/codes-formats#languages
+            // for the list of language criteria IDs.
+            campaign.LanguageConstants.Add(ResourceNames.LanguageConstant(1000)); /* English */
+
+            KeywordPlanCampaignOperation operation = new KeywordPlanCampaignOperation()
+            {
+                Create = campaign
+            };
+
+            // Add the campaign.
+            MutateKeywordPlanCampaignsResponse response =
+                serviceClient.MutateKeywordPlanCampaigns(customerId.ToString(),
+                    new KeywordPlanCampaignOperation[] { operation });
+
+            // Display the result.
+            String planCampaignResource = response.Results[0].ResourceName;
+            Console.WriteLine($"Created campaign for keyword plan: {planCampaignResource}.");
+            return planCampaignResource;
+        }
+
+        /// <summary>
+        /// Creates the ad group for the keyword plan.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        /// <param name="planCampaignResource">The resource name of the campaign under which the
+        /// ad group is created.</param>
+        /// <returns>The newly created ad group resource.</returns>
+        private string CreateKeywordPlanAdGroup(GoogleAdsClient client, long customerId,
```

Comment:
Is Ad Group a necessary resource for keyword planner to work? If not, it may be clearer for the comment to reflect that as an optional input.

Reply:
AdGroup is a required resource.